Feature/raytrace objects

dmytrozykov opened the pull request.

dmytrozykov committed:
added ray and hit structs

dmytrozykov committed:
added basic shapes structs and intersection algorithms

dmytrozykov committed:
removed point addition

Removed point addition (Point + Point), instead there is new constructor for vectors that takes start and end points. 
Added Normal alias.
Added helper degrees helper functions.

h1laryz commented on include/objects/disk.h:
```diff
@@ -0,0 +1,22 @@
+
+#include "object.h"
+
+namespace objects
```
objects is norm
but we can change this namespace to obj also
i like both

h1laryz commented on include/objects/disk.h:
```diff
@@ -0,0 +1,22 @@
+struct Disk : Object
+{
+    float radius = 1.0f;
+    lmath::Normal normal;
```
normal; -> normal{}; because it doesn't have constructor that takes 0 params, so all the variables in structure with {} will be zero by default

dmytrozykov committed:
refactor

fixed initializations
fixed compilation warning flags
renamed namespace objects to obj
